airmozilla/popcorn/tests/test_views.py

```diff
@@ -147,3 +157,140 @@ def test_popcorn_editor(self):
 
         response = self.client.get(url, {'slug': event.slug})
         eq_(response.status_code, 200)
+
+    @mock.patch('airmozilla.manage.vidly.urllib2')
+    def test_vidly_webhook(self, p_urllib2):
+        xml_string = SAMPLE_MEDIA_RESULT_SUCCESS
+        success_xml = xmltodict.parse(xml_string)
+
+        url = reverse('popcorn:vidly_webhook')
+        task = success_xml['Response']['Result']['Task']
+        tag = task['MediaShortLink']
+        file_url = task['SourceFile']
+
+        def make_mock_request(url, querystring):
+            return mock.MagicMock()
+
+        def mocked_urlopen(request):
+            xml_string = get_custom_XML(
+                tag=tag,
+                status='Finished',
+                private='false'
+            )
+            return StringIO(xml_string)
+
+        p_urllib2.Request.side_effect = make_mock_request
+        p_urllib2.urlopen = mocked_urlopen
+
+        event = Event.objects.get(title='Test event')
+        event.template.name = 'this is a vid.ly video'
+        event.template.save()
+
+        event.template_environment = {'tag': tag}
+        event.save()
+
+        VidlySubmission.objects.create(
+            event=event,
+            url=file_url,
+            tag=tag,
+            token_protection=False
+        )
+
+        response = self.client.post(url, {'xml': xml_string})
+        eq_(response.status_code, 200)
+        eq_('OK\n', response.content)
```

Comment:
You might want to check that the `VidlySubmission` that was created on line 192 changed. 
E.g. First change line 192 to `vidly_submission = VidlySubmission.objects.create(`
then do something like this:
```python
vidly_submission = VidlySubmission.objects.get(id=vidly_submission.id)
ok_(vidly_submission.finished)
```